### PR TITLE
use JsonEditor in MessageCreator

### LINKED ISSUE #1867: Use <JsonEditor> in <MessageCreator>
I've modified `<JsonEditor>` to allow external modules to modify the template.  This allows, for example, the dropdown list of assets in a form to be created from the actual assets in the game.

But, `<MessageCreator>` uses it's own `new JSONEditor()` code to create the editor control.

We should switch this class to use `<JsonEditor>` instead.

Note: when we do this, we will need to modify `<JsonEditor>` to make the `messageContent` and `messageId` optional.  When `messageId` is missing, we should provide a `_.uniqueID` value for it in `JsonEditor`.  **Update:** - hold on, we probably won't create the `_,.uniqId` - because that comes from PouchDb when we push the new document.

### PR DESCRIPTION
**Pull request by lilitkarapetyan**
## 🧰 Issue

Fixes #1867

## 🔨Work carried out:

- [x] Tests pass

## Confirmations

- [ ] I have chosen reviewers for my PR.
- [ ] I have assigned myself to this PR.
- [ ] I have chosen an appropriate label for the PR.
- [ ] I have completed the mandatory sections of this document.
- [ ] I have deleted any unused sections.
- [ ] I confirm that I have checked for required README updates and acted accordingly.

### CODE REVIEW

**Comment by IanMayo** [comment]
Thanks @lilitkarapetyan - DeepScan has found a couple of small problems.

File: packages/components/src/local/form-elements/message-creator/index.tsx [hunk 1]
```diff
@@ -113,153 +108,30 @@ const MessageCreator: React.FC<PropTypes> = ({
...
-      if (formValue) return editor && editor.setValue(formValue)
-    }, 10)
-
-    return (): void => clearTimeout(valueTimer)
-  }, [editor])
-
-  /**
-   * helper function to for validation Datetime or Date or Time props of json
-   */
-  const configCommonValidation = (
```
Comment (IanMayo) [inline packages/components/src/local/form-elements/message-creator/index.tsx:159]: @lilitkarapetyan  - it looks like we're losing a block of functionality here.    I guess we have to move these custom editors, and custom validators to `JSON-Editor`.  Aah, unless they are there already :-)

Comment (lilitkarapetyan) [reply packages/components/src/local/form-elements/message-creator/index.tsx:159]: I will work on it

**Comment by lilitkarapetyan** [comment]
> Thanks @lilitkarapetyan - DeepScan has found a couple of small problems.

it looks like we're losing a block of functionality here. I guess we have to move these custom editors, and custom validators to JSON-Editor. Aah, unless they are there already :-)

> Thanks @lilitkarapetyan - DeepScan has found a couple of small problems.

I will fix the problems

**Comment by lilitkarapetyan** [comment]
> Thanks @lilitkarapetyan - DeepScan has found a couple of small problems.

I will fix the problems

**Comment by IanMayo** [comment]
Thanks @lilitkarapetyan  - I've re-enabled the `story` for `json-editor`, so I can check the date support.  But, nothing is shown.

One other thing.  We still have instances where `JSONEditor()` is called.  I hope these are legacy code that we can delete.  Aah, `json-schema-editor` can probably stay, since that's a very specialised use of the JSONEditor.  Here are the other instances:

image


So, two changes, please:
1. Get the story working
2. Remove other calls to `JsonEditor()` - to be sure we're just using the one component.

**Comment by lilitkarapetyan** [comment]
> Thanks @lilitkarapetyan - I've re-enabled the `story` for `json-editor`, so I can check the date support. But, nothing is shown.
> 
> One other thing. We still have instances where `JSONEditor()` is called. I hope these are legacy code that we can delete. Aah, `json-schema-editor` can probably stay, since that's a very specialised use of the JSONEditor. Here are the other instances:
> 
> image
> 
> So, two changes, please:
> 
>     1. Get the story working
> 
>     2. Remove other calls to `JsonEditor()` - to be sure we're just using the one component.
Hello Ian, I will work on it

**Comment by IanMayo** [comment]
Hello @lilitkarapetyan ,  `MessageCreator` takes a prop called `customizeTemplate`.  This modifies the template, potentially setting default values, or populating drop-downs.

Here is an example of `customizeTemplate` from `SupportPanel`:
image

If one is supplied, we need to apply the `customizeTemplate` function to the selected template in `MessageCreator`.

**Review by IanMayo — APPROVED** [review]
Nice package of work. Thanks @lilitkarapetyan

**Pull request merged**